### PR TITLE
[SPARK-36096][CORE] Grouping exception in core/resource

### PR DESCRIPTION
**Pull request by dgd-contributor**
### What changes were proposed in this pull request?
This PR group exception messages in core/src/main/scala/org/apache/spark/resource

### Why are the changes needed?
It will largely help with standardization of error messages and its maintenance.

### Does this PR introduce _any_ user-facing change?
No. Error messages remain unchanged.

### How was this patch tested?
No new tests - pass all original tests to make sure it doesn't break any existing behavior.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 1]
```diff
@@ -141,4 +144,151 @@ object SparkCoreErrors {
   def mustSpecifyCheckpointDirError(): Throwable = {
     new SparkException("Checkpoint dir must be specified.")
   }
+
+  def acquireAnAddressNotExistError(resourceName: String, address: String): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:148]: acquireNonExistingAddressError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 2]
```diff
@@ -141,4 +144,151 @@ object SparkCoreErrors {
...
+    new SparkException(s"Try to acquire an address that doesn't exist. $resourceName " +
+      s"address $address doesn't exist.")
+  }
+
+  def acquireAnAddressNotAvailableError(resourceName: String, address: String): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:153]: acquireUnavailableAddressError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 3]
```diff
@@ -141,4 +144,151 @@ object SparkCoreErrors {
...
+    new SparkException("Try to acquire an address that is not available. " +
+      s"$resourceName address $address is not available.")
+  }
+
+  def releaseAnAddressNotExistError(resourceName: String, address: String): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:158]: releaseNonExistingAddressError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 4]
```diff
@@ -141,4 +144,151 @@ object SparkCoreErrors {
...
+    new SparkException(s"Try to release an address that doesn't exist. $resourceName " +
+      s"address $address doesn't exist.")
+  }
+
+  def releaseAnAddressNotAssignedError(resourceName: String, address: String): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:163]: releaseUnassignedAddressError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 5]
```diff
@@ -141,4 +144,151 @@ object SparkCoreErrors {
...
+    new SparkException(s"Try to release an address that is not assigned. $resourceName " +
+      s"address $address is not assigned.")
+  }
+
+  def resourceScriptNotExistError(scriptFile: File, resourceName: String): Throwable = {
+    new SparkException(s"Resource script: $scriptFile to discover $resourceName " +
+      "doesn't exist!")
+  }
+
+  def expectUseResourceButNotSpecifyADiscoveryScriptError(resourceName: String): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:173]: noDiscoveryScriptSpecifiedError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 6]
```diff
@@ -141,4 +144,151 @@ object SparkCoreErrors {
...
+    new SparkException(s"User is expecting to use resource: $resourceName, but " +
+      "didn't specify a discovery script!")
+  }
+
+  def runningOtherResourceError(
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:178]: discoveryScriptNameMismatchError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 7]
```diff
@@ -141,4 +144,151 @@ object SparkCoreErrors {
...
+      script: Optional[String],
+      name: String,
+      resourceName: String): Throwable = {
+    new SparkException(s"Error running the resource discovery script ${script.get}: " +
+      s"script returned resource name ${name} and we were expecting $resourceName.")
+  }
+
+  def ParsingJsonToResourceInformationError(
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:186]: failToParseJsonToResourceInformationError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 8]
```diff
@@ -141,4 +144,151 @@ object SparkCoreErrors {
...
+      json: String,
+      exampleJson: String,
+      e: Throwable): Throwable = {
+    new SparkException(s"Error parsing JSON into ResourceInformation:\n$json\n" +
+      s"Here is a correct example: $exampleJson.", e)
+  }
+
+  def ParsingJsonToResourceInformationError(json: JValue, e: Throwable): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:194]: failToParseJsonToResourceInformationError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 9]
```diff
@@ -141,4 +144,151 @@ object SparkCoreErrors {
...
+    new SparkException(s"Error parsing JSON into ResourceInformation:\n$json\n", e)
+  }
+
+  def resourceNotExistInProfileIdError(resource: String, id: Int): Throwable = {
+    new SparkException(s"Resource $resource doesn't exist in profile id: $id")
+  }
+
+  def conditionOfExecutorResourceError(rName: String, num: Long, taskReq: Double): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:202]: executorResourceLessThanTaskResourceRequestError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 10]
```diff
@@ -141,4 +144,151 @@ object SparkCoreErrors {
...
+    new SparkException(s"The executor resource: $rName, amount: ${num} " +
+      s"needs to be >= the task resource request amount of $taskReq")
+  }
+
+  def noExecutorResourceConfigError(str: String): Throwable = {
+    new SparkException("No executor resource configs were not specified for the " +
+      s"following task configs: ${str}")
+  }
+
+  def resourceProfileSupportedError(): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:212]: resourceProfilesUnsupportedError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 11]
```diff
@@ -141,4 +144,151 @@ object SparkCoreErrors {
...
+    new SparkException("ResourceProfiles are only supported on YARN and Kubernetes " +
+      "with dynamic allocation enabled.")
+  }
+
+  def resourceProfileIdNotFoundError(rpId: Int): Throwable = {
+    new SparkException(s"ResourceProfileId $rpId not found!")
+  }
+
+  def specifyConfigOfResourceError(str: String): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:221]: noAmountSpecifiedForResourceError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 12]
```diff
@@ -141,4 +144,151 @@ object SparkCoreErrors {
...
+    new SparkException(s"You must specify an amount for ${str}")
+  }
+
+  def specifyAnAmountConfigForResourceError(
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:225]: noAmountConfigSpecifiedForResourceError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 13]
```diff
@@ -141,4 +144,151 @@ object SparkCoreErrors {
...
+      key: String,
+      componentName: String,
+      RESOURCE_PREFIX: String): Throwable = {
+    new SparkException(s"You must specify an amount config for resource: $key " +
+      s"config: $componentName.$RESOURCE_PREFIX.$key")
+  }
+
+  def conditionOfResourceAmountError(doubleAmount: Double): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:233]: invalidResourceAmountError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 14]
```diff
@@ -141,4 +144,151 @@ object SparkCoreErrors {
...
+    new SparkException(
+      s"The resource amount ${doubleAmount} must be either <= 0.5, or a whole number.")
+  }
+
+  def specifyAmountForResourceError(str: String): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:238]: Can be combined with
```
def specifyConfigOfResourceError(str: String): Throwable = {
    new SparkException(s"You must specify an amount for ${str}")
  }
```

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 15]
```diff
@@ -141,4 +144,151 @@ object SparkCoreErrors {
...
+    new SparkException(s"You must specify an amount for ${str}")
+  }
+
+  def tasksSupportFractionalResourceError(componentName: String): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:242]: fractionalResourcesUnsupportedError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 16]
```diff
@@ -141,4 +144,151 @@ object SparkCoreErrors {
...
+    new SparkException(
+      s"Only tasks support fractional resources, please check your $componentName settings")
+  }
+
+  def ParsingResourceFileError(resourcesFile: String, e: Throwable): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:247]: failToParseResourceFileError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 17]
```diff
@@ -141,4 +144,151 @@ object SparkCoreErrors {
...
+    new SparkException(s"Error parsing resources file $resourcesFile", e)
+  }
+
+  def conditionOfTheNumberOfCoresPerExecutorError(execCores: Int, taskCpus: Int): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:251]: numCoresPerExecutorLessThanNumCPUsPerTaskError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 18]
```diff
@@ -141,4 +144,151 @@ object SparkCoreErrors {
...
+    new SparkException(s"The number of cores per executor (=$execCores) has to be >= " +
+      s"the number of cpus per task = $taskCpus.")
+  }
+
+  def nonOfTheDiscoveryPluginsReturnedResourceInfoError(str: String): Throwable = {
+    new SparkException(s"None of the discovery plugins returned ResourceInformation for " +
+      s"${str}")
+  }
+
+  def adjustConfigurationError(
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:261]: adjustConfigurationofCoresError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 19]
```diff
@@ -141,4 +144,151 @@ object SparkCoreErrors {
...
+      s"""
+        |The configuration of cores (exec = ${cores}
+        |task = ${taskCpus}, runnable tasks = ${resourceNumSlots}) will
+        |result in wasted resources due to resource ${limitingResource} limiting the
+        |number of runnable tasks per executor to: ${maxTaskPerExec}. Please adjust
+        |your configuration.
+      """.stripMargin.replaceAll("\n", " "))
+  }
+
+  def adjustConfigurationError(
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:277]: adjustConfigurationOfResourceError

**Comment by dgd-contributor** [comment]
cc @mridulm @cloud-fan can you review this

**Comment by dgd-contributor** [comment]
Thanks for your reviews. As we and @HyukjinKwon discussed contributing by sharing accounts, we will close this PR and recreate another by individual account. Please see PR [#34192](https://github.com/apache/spark/pull/34192)

**Pull request closed without merging**